core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2857,6 +2857,108 @@ private void checkSarg(String message, Sarg sarg,
         is(2), is("Sarg[[3..8], [10..20]]"));
   }
 
+  /** Tests {@link Sarg#numDistinctVals(RelDataType)}. */
+  @Test void testSargNdv() {
+    RelDataType nonNullableInt = typeFactory.createSqlType(SqlTypeName.INTEGER);
+    RelDataType nullableInt = typeFactory.createTypeWithNullability(nonNullableInt, true);
+
+    RelDataType nonNullableFloat = typeFactory.createSqlType(SqlTypeName.FLOAT);
+    RelDataType nullableFloat = typeFactory.createTypeWithNullability(nonNullableFloat, true);
+
+    // point: 10 (non-nullable)
+    Sarg sarg = Sarg.of(false,
+        ImmutableRangeSet.of(Range.closed(10, 10)));
+    assertThat(sarg.numDistinctVals(nonNullableInt), is(1.0));
```

Comment:
Could you please factor this to a helper method that receives `sarg`, `type`, and the expected matcher?

Then the error message could include `sarg + ".numDistinctVals(" + type + ")"` so the developers could see the inputs that produced invalid output.

The current assertion failures would look like "expected 1.0 got 1.5", and it would be hard to tell what is going on.


An alternative option is to split assertions to individual tests (e.g. parameterized)

Reply:
Thank you for the good suggestion.
I have revised the code accordingly.